packages/jest-runtime/src/index.ts

```diff
@@ -1327,15 +1327,19 @@ export default class Runtime {
   }
 
   private _requireCoreModule(moduleName: string) {
-    if (moduleName === 'process') {
+    const moduleWithoutNodePrefix = moduleName.startsWith('node:')
```

Comment:
We may do not strip the prefix if the runtime supports it:
```js
process.version.slice(1, 3) < 16
semver.gt(process.version, 'v16')
```

ref: https://nodejs.org/api/modules.html#modules_core_modules

Reply:
[`94d1903` (#11331)](https://github.com/facebook/jest/pull/11331/commits/94d19038e15a481e71c02947a857a882265ef07d)